packages/commonwealth/server/migrations/20250314120000-add-user-tier.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      await queryInterface.sequelize.query(
+        `
+        ALTER TABLE "Users" ADD COLUMN "tier" INTEGER NOT NULL DEFAULT 0;
+
+        -- seed tiers based on verification status
+        UPDATE "Users"
+        SET "tier" = CASE 
+            -- TODO: find tier 4 users
+            WHEN "emailVerified" = TRUE THEN 3
```

Comment:
`emailVerified` in `Users` table doesn't actually mean the email is verified. From `magic.ts`:
> // we mark email verified so that we are OK to send update emails, but we should note that
        // just because an email comes from magic doesn't mean it's legitimately owned by the signing-in
        // user, unless it's via the email flow (e.g. you can spoof an email on Discord -> Discord allows oauth
        // sign in with unverified email addresses)

Tier 3 users would be any user that has at least 1 address where `oauth_provider` is NOT NULL _and_ one or more of the following is true:
- `oauth_email` is NOT NULL and `oauth_email_verified = true`
- `oauth_phone_number` is NOT NULL
- `oauth_username` IS NOT NULL

Reply:
`magic.ts` and the tierd middleware upgrading